pennylane/ops/qubit/observables.py

```diff
@@ -680,9 +702,9 @@ def compute_eigvals(state_vector):  # pylint: disable=arguments-differ,arguments
         >>> StateVectorProjector.compute_eigvals([0, 0, 1, 0])
         array([1, 0, 0, 0])
         """
-        w = qml.math.zeros_like(state_vector)
+        w = np.zeros(qml.math.shape(state_vector), dtype="float64")
```

Comment:
Why does this one have to be `float64`?

Reply:
fixing to check the precision of `state_vector`.

Reply:
And realizing that the lower `cast_like` was reverting all this work anyway 🤦